### ISSUE TITLE
Adding formulas to rmarkdown documents

### ISSUE THREAD

**Issue by oganm** [STAT545-UBC/Discussion#328]
By default latex formula syntax is incompatible with github's markdown syntax. The recommended way to add formulas is to add images instead as there are many services that convert latex code to an image. The code below will automate this for you.  Just add an inline R code with the \`r  latexImg("2+2") ` snytax and an image will be added in place of your formula

```
latexImg = function(latex, markdown=T){
    
    link = paste0('http://latex.codecogs.com/gif.latex?',
                  gsub('\\=','%3D',URLencode(latex)))
    
    # they need to be upper case
    link = gsub("(%..)","\\U\\1",link,perl=TRUE)
    if (!markdown){
        return(link)
    }
    return(paste0('![](',link,')'))
}
```

**Comment by jennybc** [comment]
> By default latex formula syntax is incompatible with github's markdown syntax.

Oh I wouldn't say that! But maybe we are misunderstanding each other?

It's more that latex formulas will only be rendered the way you expect and hope by MathJax once you render all the way to PDF or html, not markdown (github-flavored markdown or not).

If I really needed equations, it would be a reason to serve my document from one of those formats.

Pretty cool example of a workaround though! I didn't know about that!

**Comment by oganm** [comment]
Indeed I misspoke. I meant that while github markdown produces an html output out of the markdown file we have, it will not convert the common syntax to add formulas. This actually came up last year as well. (#102). You did link to [another potential solution](https://mikelove.wordpress.com/2015/07/01/how-to-use-latex-math-in-rmd-to-display-properly-on-github-pages/) but I never tried that one myself.

**Comment by jennybc** [comment]
Yeah I remember this. But even the solution you / I link to presumes one is targeting full-blown [GitHub Pages](https://pages.github.com), which we are not. And would not be compatible with our private repos.

This comes up now, because people like to play with math when tricking out their README. But luckily we really don't need to include many (any?) heavy equations for the rest of the course.

**Comment by leegao** [comment]
I totally hope I'm not stepping out of line here, but I stumbled upon this a while back while googling for LaTeX support in gh-flavored markdown. After a lot of Googling, I ended up hacking up a small tool that:

1. Compiles your formulas down into images and
2. Manipulates the resulting image so that it will line up with the baseline of the rest of the text (so you can add inline equations as well)

It seems to work quite well for me, and I hope that it could solve your problem as well.

https://github.com/leegao/readme2tex/blob/master/README.md

**Comment by gabrielespinosa** [comment]
Fast general Feedback. It would be nice if the *.ipynb files ``markdown`` option to display general latex test or equation could be joined with the ``*.md``, ``*.rst``, et al. files.

A fast way to obtain results of Latex ``test`` or ``equation`` results of the image could also by find in the open webpage:

Link from: [Latex equation display academic network](http://www.sciweavers.org/free-online-latex-equation-editor)

```Latex
http://www.sciweavers.org/free-online-latex-equation-editor
````
